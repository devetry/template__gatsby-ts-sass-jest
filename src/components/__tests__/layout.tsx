import * as React from "react"
import renderer from "react-test-renderer"
import * as Gatsby from "gatsby" // mocked

import Layout from "../layout"

beforeEach(() => {
  const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
  useStaticQuery.mockImplementationOnce(() => ({
    site: {
      siteMetadata: {
        title: `GatsbyJS`,
      },
    },
  }));
})

describe(`Layout`, () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Layout>
          <main>
            <h1>hello</h1>
          </main>
        </Layout>
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})