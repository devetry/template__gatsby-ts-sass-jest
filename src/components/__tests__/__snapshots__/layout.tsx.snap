// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          GatsbyJS
        </a>
      </h1>
    </div>
  </header>,
  <div>
    <main>
      <main>
        <h1>
          hello
        </h1>
      </main>
    </main>
    <footer>
      Â© 
      2021
      , Built with
       
      <a
        href="https://www.gatsbyjs.com"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
